plugins/Tools.cc

```diff
@@ -339,6 +339,91 @@ void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilept
     }
 }
 
+float HHAnalyzer::getL1TPhiGen(const Lepton& lep) {
+    if (!lep.gen_matched)
+        return getL1TPhiReco(lep);
+    float gen_pt = lep.gen_p4.Pt();
+    float gen_phi = lep.gen_p4.Phi();
+    float gen_theta = lep.gen_p4.Theta();
+    return gen_phi + lep.charge * (1. / gen_pt) * (10.48 - 5.1412 * gen_theta + 0.02308 * gen_theta * gen_theta);
+}
+
+float HHAnalyzer::getL1TPhiReco(const Lepton& lep) {
```

Comment:
couldn't you have this one taking just a p4 instead ? this would save you the code duplicate above, we need the full lepton object only to get the mc truth ?